how-to-run-a-node/validator-key-management.md

```diff
@@ -0,0 +1,186 @@
+
+### Staking Parameters
+- **Caller**: Endorser public address (the account depositing the VET)
+- **Validator**: Master key public address (from `bin/thor master-key`)
+- **Period**: Lock period in blocks (60480/129600/259200 for 7/15/30 days)
+- **Value**: Amount of VET to stake (minimum 25M VET for validators)
+
+### Staking Workflow
+1. Generate/obtain master key: `bin/thor master-key`
+2. Ensure node is fully synced: `bin/thor --network main`
+3. Use the master key address as the "Validator" parameter in staking contract
+4. Use your VET-holding account as the "Caller/Endorser" address
+5. Execute staking transaction with proper parameters
+
+## Security Best Practices
+
+### File Permissions and Backup
+```bash
+# Ensure key files have restricted permissions
+chmod 600 ~/.org.vechain.thor/master.key
+chmod 600 ~/.org.vechain.thor/p2p.key
+
+# Export keys immediately after generation
+bin/thor master-key --export > backup-keystore.json
+```
+
+### Key Security Guidelines
+- **Store backups securely** offline in multiple locations
+- **Test recovery process** with backups before going live
+- **Use strong passphrases** for encrypted exports
+- **Separate concerns**: Use different addresses for staking (endorser) and validation (master key)
+- **Regular backups**: Export keys after any changes or before major updates
+- **Monitor key usage**: Track validator address activity on block explorers
+
+## Recovery Scenarios
+
+### Lost Master Key
+If you lose your master key:
+1. You'll need to generate a new key (loses previous validator identity)
+2. Re-stake with the new master key address
+3. Previous stake remains locked to old address until maturity
+4. Update any monitoring or management systems with new validator address
+
+### Node Migration
+When moving to a new server:
+1. **Export** master key from old node: `bin/thor master-key --export > migration-keystore.json`
+2. **Import** on new node: `cat migration-keystore.json | bin/thor master-key --import`
+3. **Verify** address matches: `bin/thor master-key`
+4. **Test** node operation before decommissioning old node
+
+### Configuration Directory Issues
+- **Default location**: `~/.org.vechain.thor/` on Linux/macOS
+- **Override with**: `--config-dir` flag
+- **Ensure proper permissions** on directory and files
+- **Verify path accessibility** before starting production operations
+
+## Advanced Key Management
+
+### Key Rotation for New Staking Periods
+1. **Generate new master key** before current stake period ends
+2. **Test new key** with node operation
+3. **Backup both keys** until transition is complete
+4. **Re-stake with new key** when ready
+5. **Decommission old key** after stake maturity
+
+### Multiple Validator Setup
+Each validator node requires:
+- Unique master key
+- Separate configuration directory
+- Distinct network ports
+- Individual staking transactions
+
+## Troubleshooting
+
+### Common Issues and Solutions
```

Comment:
Would also suggest mentioning solution for docker users.

Reply:
That might be tricker as it depends on the volume mount usage.

Reply:
Right, just leave it there